src/platform/i386/vm.c

```diff
@@ -57,7 +60,12 @@ kern_setup_image(void)
 {
 	unsigned long i, j;
 	paddr_t       kern_pa_start, kern_pa_end;
+	int cpu_id = get_cpuid();
 
+	if (cpu_id > 0) {
```

Comment:
Perhaps not call this in for APs, instead directly call `chal_cpu_init()`?

Reply:
Yeah, good point.